mps-model-adapters-plugin/src/test/kotlin/org/modelix/model/mpsadapters/ReplaceNodeTest.kt

```diff
@@ -1,40 +1,108 @@
 package org.modelix.model.mpsadapters
 
-import org.junit.Ignore
+import jetbrains.mps.smodel.SNode
+import jetbrains.mps.smodel.adapter.MetaAdapterByDeclaration
 import org.modelix.model.api.BuiltinLanguages
 import org.modelix.model.api.ConceptReference
 import org.modelix.model.api.INode
 import org.modelix.model.api.IReplaceableNode
 
-@Ignore("Replacing a node through MPS-model-adapters is broken. See MODELIX-920")
 class ReplaceNodeTest : MpsAdaptersTestBase("SimpleProject") {
 
-    fun testReplaceNode() {
-        readAction {
-            assertEquals(1, mpsProject.projectModules.size)
-        }
+    private val sampleConcept = MetaAdapterByDeclaration.asInstanceConcept(
+        MPSConcept.tryParseUID(BuiltinLanguages.jetbrains_mps_lang_core.BaseConcept.getUID())!!.concept,
+    )
+
+    fun `test replace node with parent and module (aka regular node)`() = runCommandOnEDT {
+        val rootNode = getRootUnderTest()
+        val nodeToReplace = rootNode.allChildren.first() as IReplaceableNode
+        val nodesToKeep = rootNode.allChildren.drop(1)
+        val oldProperties = nodeToReplace.getAllProperties().toSet()
+        check(oldProperties.isNotEmpty()) { "Test should replace node with properties." }
+        val oldReferences = nodeToReplace.getAllReferenceTargetRefs().toSet()
+        check(oldReferences.isNotEmpty()) { "Test should replace node with references." }
+        val oldChildren = nodeToReplace.allChildren.toList()
+        check(oldChildren.isNotEmpty()) { "Test should replace node with children." }
+        val newConcept = ConceptReference("mps:f3061a53-9226-4cc5-a443-f952ceaf5816/1083245097125")
+
+        val newNode = nodeToReplace.replaceNode(newConcept)
+
+        assertEquals(listOf(newNode) + nodesToKeep, rootNode.allChildren.toList())
+        assertEquals(oldProperties, newNode.getAllProperties().toSet())
+        assertEquals(oldReferences, newNode.getAllReferenceTargetRefs().toSet())
+        assertEquals(oldChildren, newNode.allChildren.toList())
+        assertEquals(newConcept, newNode.getConceptReference())
```

Comment:
Also assert that the ID of the node is the same as before.
This will fail because `SModel.createNode` will not use the provided ID if it already exists.

Reply:
Done